### PR TITLE
feat(validations): add support to bind dir validators

### PR DESCRIPTION
**Pull request by kara**
This PR builds on @tolemac's original PR to finish adding support for dir validators.   He's done most of the work here; just putting on finishing touches.

r: @IgorMinar, @vsavkin 

Summary: Previously, it wasn't possible to bind to validators like `required`, `minlength`, `maxlength`, and `pattern` in the DOM.  If you did the following:

``` ts
<input name="first" ngModel [required]="isRequired">
```

...regardless of the value of `isRequired`, validation would be run as if required were set.  This is because the built-in validator directives weren't set up to respond to inputs and to re-run validation when those inputs changed.   Now the above should work!

### CODE REVIEW

File: modules/@angular/forms/src/directives/ng_control.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export abstract class NgControl extends AbstractControlDirective {
 
   get validator(): ValidatorFn { return <ValidatorFn>unimplemented(); }
   get asyncValidator(): AsyncValidatorFn { return <AsyncValidatorFn>unimplemented(); }
+  get rawValidators(): any[] { return unimplemented(); }
```
Comment (vsavkin) [inline modules/@angular/forms/src/directives/ng_control.ts]: why `any[]`?

Comment (kara) [inline modules/@angular/forms/src/directives/ng_control.ts]: Oh, I think it should be `Validator[]`. I can update.

File: modules/@angular/forms/src/directives/ng_model.ts [hunk 2]
```diff
@@ -105,6 +105,8 @@ export class NgModel extends NgControl implements OnChanges,
 
               get validator(): ValidatorFn { return composeValidators(this._validators); }
 
+              get rawValidators(): Array<Validator|ValidatorFn> { return this._validators || []; }
```
Comment (vsavkin) [inline modules/@angular/forms/src/directives/ng_model.ts]: Why do we include only sync validators here?

**Comment by vsavkin** [comment]
We're doing it this way because we don't it to be a breaking change? I find it a bit weird to have three properties instead of two.

**Pull request closed without merging**